src/coreclr/jit/lower.cpp

```diff
@@ -1261,13 +1261,7 @@ void Lowering::LowerArg(GenTreeCall* call, CallArg* callArg, bool late)
     DISPNODE(arg);
     assert(arg->IsValue());
 
-    var_types type = arg->TypeGet();
-
-    if (varTypeIsSmall(type))
-    {
-        // Normalize 'type', it represents the item that we will be storing in the Outgoing Args
-        type = TYP_INT;
```

Comment:
Just that I understand, was there an inconsistency in the way we set type here to `TYP_INT` but not doing the same in `ToScalar()`?

Reply:
Got it. So is the assert reported in https://github.com/dotnet/runtime/issues/71360 is the only one that checks for this or do we have other places we make sure that type is never one of `TYP_UINT` or `TYP_ULONG`?

Reply:
Sounds good to me.

Reply:
> and maybe some SIMD base types

Its most SIMD base types in practice. The SIMD base type should always be the exact type required for the underlying instruction. That includes correctly tracking the sign and size. This is important because there are many SIMD instructions that differ based on size/type and will do the wrong thing otherwise.

It was definitely a bug that the `gtType` wasn't the right thing, however.